src/main/java/de/tum/in/www1/artemis/service/TitleCacheEvictionService.java

```diff
@@ -41,13 +43,22 @@ public class TitleCacheEvictionService implements PostUpdateEventListener, PostD
 
     private final CacheManager cacheManager;
 
+    private final EntityManagerFactory entityManagerFactory;
+
     public TitleCacheEvictionService(EntityManagerFactory entityManagerFactory, CacheManager cacheManager) {
         this.cacheManager = cacheManager;
+        this.entityManagerFactory = entityManagerFactory;
+    }
 
+    @EventListener(ApplicationReadyEvent.class)
+    public void applicationReady() {
         var eventListenerRegistry = entityManagerFactory.unwrap(SessionFactoryImpl.class).getServiceRegistry().getService(EventListenerRegistry.class);
-        eventListenerRegistry.appendListeners(EventType.POST_UPDATE, this);
-        eventListenerRegistry.appendListeners(EventType.POST_DELETE, this);
-        log.debug("Registered Hibernate listeners");
+        if (eventListenerRegistry != null) {
+            eventListenerRegistry.appendListeners(EventType.POST_UPDATE, this);
+            eventListenerRegistry.appendListeners(EventType.POST_DELETE, this);
+            log.debug("Registered Hibernate listeners");
+        }
+        log.warn("Could not register Hibernate listeners because the EventListenerRegistry is null. This is likely due to a misconfiguration of the entity manager factory.");
```

Comment:
```suggestion
        } else {
            log.warn("Could not register Hibernate listeners because the EventListenerRegistry is null. This is likely due to a misconfiguration of the entity manager factory.");
        }
```